pages/[id]/index.js

```diff
@@ -1,16 +1,58 @@
 import initialPictures from "@/store";
 import PostingDetails from "@/components/PostingDetails";
 import { useRouter } from "next/router";
+import CommentForm from "@/components/CommentForm";
+import { useState } from "react";
+import { uid } from "uid";
+import CommentList from "@/components/CommentList";
 
 export default function PostDetailsPage() {
+  const [pictures, setPictures] = useState(initialPictures);
   const router = useRouter();
   const { id } = router.query;
 
-  const picture = initialPictures.find((picture) => {
+  // getting the requested picture
+  const selectedPicture = pictures.find((picture) => {
     return picture.id === id;
   });
 
-  if (picture) {
-    return <PostingDetails picture={picture} />;
+  // adding comment
+  function handleAddComment(comment) {
+    setPictures(
+      pictures.map((picture) => {
+        if (selectedPicture.id === picture.id) {
+          picture.comments.push({ commentId: uid(), commentContent: comment });
```

Comment:
you should not mutate entries in a state. We always want to work with copies. Like this:

```suggestion
            return {...picture, comments: [...picture.comments, { commentId: uid(), commentContent: comment }, ], };
```

unless you work with a library like use-immer